### PR TITLE
Serialize projectionDefinition to arrays

### PR DESCRIPTION
**Pull request by birkskyum**
This is how it would look to serialize to arrays like color does.

## Launch Checklist

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/expression/types/projection_definition.ts [hunk 1]
```diff
@@ -7,10 +7,11 @@ export class ProjectionDefinition {
         this.from = from;
         this.to = to;
         this.transition = transition;
+        
     }
 
     static interpolate(from: string, to: string, t: number) {
-        return new ProjectionDefinition(from, to, t);
+        return new ProjectionDefinition(from, to, t).toJSON();
```
Comment (HarelM) [inline src/expression/types/projection_definition.ts:14]: I think interpolate should return the same type of object and not a string...

**Comment by HarelM** [comment]
I'm failing to see the motivation behind this PR to be honest...

File: src/expression/types/projection_definition.ts (hunk 1 above)
Comment (birkskyum) [reply src/expression/types/projection_definition.ts:14]: Okay,

Just to clarify, this does return an array, not a string.

**Comment by birkskyum** [comment]
@HarelM , thanks for being honest with it, I'm fine with either.

> I do see that color does get serialized to array somehow (saw it in test/integration/expression/tests/interpolate-hcl/linear/test.json test), but I couldn't figure out where it happens.

My main motivation here was to ensure that we could freely decide on API design, if we serialize to arrays or class objects, instead of having the choice made for us due to technical limitations.

**Pull request closed without merging**